resources/js/Pages/Project/Tasks/Partials/ProjectTasks.vue

```diff
@@ -0,0 +1,301 @@
+              {{ $t('app.cancel') }}
+            </a>
+          </div>
+        </form>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import Checkbox from '@/Shared/EditableCheckbox';
+import TextArea from '@/Shared/TextArea';
+import LoadingButton from '@/Shared/LoadingButton';
+import SelectBox from '@/Shared/Select';
+
+export default {
+  components: {
+    TextArea,
+    Checkbox,
+    LoadingButton,
+    SelectBox,
+  },
+
+  props: {
+    tasks: {
+      type: Array,
+      default: null,
+    },
+    project: {
+      type: Object,
+      default: null,
+    },
+    taskList: {
+      type: Object,
+      default: null,
+    },
+    members: {
+      type: Array,
+      default: null,
+    },
+  },
+
+  data() {
+    return {
+      localTasks: [],
+      loadingState: '',
+      addTaskMode: false,
+      taskToEdit: 0,
+      form: {
+        assignee_id: null,
+        title: null,
+        description: null,
+        task_list_id: null,
+        errors: [],
+      },
+    };
+  },
+
+  mounted() {
+    if (this.tasks) {
+      this.localTasks= this.tasks;
+    }
+
+    if (this.taskList) {
+      this.form.task_list_id = this.taskList.id;
+    }
+  },
+
+  methods: {
+    displayAddTask() {
+      this.addTaskMode = true;
+      this.form.title = null;
+      this.form.description = null;
+
+      this.$nextTick(() => {
+        this.$refs['newTaskItem'].$refs['input'].focus();
+      });
+    },
+
+    showEditTask(task) {
+      this.taskToEdit = task.id;
+      this.form.title = task.title;
+      if (task.assignee) {
+        this.form.assignee_id = {
+          value: task.assignee.id,
+          label: task.assignee.name,
+        };
+      }
+
+      // this is really barbaric, but I need to do this to
+      // first: target the TextInput with the right ref attribute
+      // second: target within the component, the refs of the input text
+      // this is because we try here to access $refs from a child component
+      this.$nextTick(() => {
+        this.$refs[`task${task.id}`][0].$refs['input'].focus();
+      });
+    },
+
+    store() {
+      this.loadingState = 'loading';
+
+      if (this.form.assignee_id) {
+        this.form.assignee_id = this.form.assignee_id.value;
+      }
+
+      axios.post(`/${this.$page.props.auth.company.id}/projects/${this.project.id}/tasks`, this.form)
+        .then(response => {
+          this.addTaskMode = false;
+          this.localTasks.push(response.data.data);
+          this.form.title = null;
+          this.form.description = null;
+          this.form.assignee_id = null;
+          this.loadingState = null;
+          flash(this.$t('project.task_create_success'), 'success');
+        })
+        .catch(error => {
+          this.loadingState = null;
+          this.form.errors = error.response.data;
+        });
+    },
+
+    toggle(id) {
+      axios.post(`/${this.$page.props.auth.company.id}/projects/${this.project.id}/tasks/${id}/toggle`)
```

Comment:
toggle should not be a "post" but a "put" instead.

Reply:
👍